### PR TITLE
feat: add batchSaveReplayResult method

### PR DESCRIPTION
**Pull request by pangdayuan1**
1. add batchSaveReplayResult method
2. optimized code. extract the batch save action that interacts with the agent

### CODE REVIEW

File: arex-storage-model/src/main/java/com/arextest/model/replay/CompareReplayResult.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.arextest.model.replay;
+
+import com.arextest.model.mock.MockCategoryType;
+import lombok.Data;
+
+/**
+ * created by xinyuan_wang on 2023/06/04
+ */
+@Data
+public class CompareReplayResult {
```
Comment (lucas-myx) [inline arex-storage-model/src/main/java/com/arextest/model/replay/CompareReplayResult.java:10]: consider if need attributes, such as sql and parameter compare

File: arex-storage-model/src/main/java/com/arextest/model/replay/CompareReplayResult.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   */
+  private String testMsg;
+  private long recordTime;
+  private long replayTime;
+  private String appId;
+  /**
+   * Check whether the recording and playback request messages are consistent.
+   * If they are consistent, only baseMsg needs to be passed.
+   */
+  private boolean sameMsg;
```
Comment (mr3) [inline arex-storage-model/src/main/java/com/arextest/model/replay/CompareReplayResult.java:31]: CompareRelationResult

// dependency request/entrypint response
recordMessage
replayMessage

sameMessage

File: arex-storage-web-api/src/main/java/com/arextest/storage/cache/CacheKeyUtils.java [hunk 3]
```diff
@@ -102,6 +104,14 @@ public static byte[] buildMatchedRecordInstanceIdsKey(MockCategoryType category,
         .array();
   }
 
+  public static byte[] buildAgentVersionKey(String replayId) {
+    return (AGENT_VERSION_REDIS_KEY + replayId).getBytes(StandardCharsets.UTF_8);
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/cache/CacheKeyUtils.java:108]: Is there a compatibility issue with this version-key?

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/AgentWorkingService.java [hunk 4]
```diff
@@ -57,16 +57,40 @@ public class AgentWorkingService {
...
+    return repositoryWriter != null && repositoryWriter.save(item);
+  }
+
+  public <T extends Mocker> boolean batchSaveRecord(@NotNull List<T> list) {
+    if (CollectionUtils.isEmpty(list)) {
+      return false;
+    }
+
+    for (T item : list) {
+      if (!prepareAndDispatchRecord(item)) {
```
Comment (lucas-myx) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/AgentWorkingService.java:74]: Is this method mainly used to intercept some operations when saving mockers？like: sql parse?

File: arex-storage-web-api/src/main/java/com/arextest/storage/metric/AgentWorkingMetricService.java [hunk 5]
```diff
@@ -55,10 +56,27 @@ public <T extends Mocker> boolean saveRecord(@NotNull T item) {
...
+    if (CollectionUtils.isEmpty(metricListeners)) {
+      return agentWorkingService.batchSaveRecord(list);
+    }
+
+    long startTimeNanos = System.nanoTime();
+    boolean result = agentWorkingService.batchSaveRecord(list);
+    long totalTimeNanos = System.nanoTime() - startTimeNanos;
+
+    recordEntryTime(METHOD_NAME_BATCH_SAVE, (AREXMocker) list.get(0), nanosToMillis(totalTimeNanos));
+    return result;
```
Comment (lucas-myx) [inline arex-storage-web-api/src/main/java/com/arextest/storage/metric/AgentWorkingMetricService.java:77]:
```
if (CollectionUtils.isEmpty(metricListeners)) {
  recordEntryTime(METHOD_NAME_BATCH_SAVE, (AREXMocker) list.get(0), nanosToMillis(totalTimeNanos));
}
```

**Pull request closed without merging**